### DISCUSSION TITLE
Power limit with Shelly meter

### DISCUSSION THREAD

**Discussion by gigasimo** [WillCodeForCats/solaredge-modbus-multi#593] [General]
Hi, I need to set the Site Limit dinamically (every 10-15 sec) to avoid to export power to the grid.
I haven't a SolarEdge Meter but I use a Shelly EM3 in Home Assistant to measure grid import/export.
There is a way in HA to update Site Limit value to avoid grid export?

**Comment by dnamat88** [comment]
upvote. i want to do "derating" power to cut off high voltage and protection of inverter. I need little progressive cuts and reverse slowly to max power when i have good voltage.

**Comment by cyberkryten** [comment]
NOTE:Not sure if you have a battery and if any/all of these commands are available to you if you don't.

If you look at the Inverter Settings, there's a Configuration section which has "I1_Site_Limit" that you can use to set and export limit.

I attempted to use this when I had issues with the Backup Gateway comms, meaning the meters were unavailable but found that changing this value caused a full inverter reset meaning it was down for 60-90 seconds, so it probably isn't viable for your use case

You may be able to use "Active Power Limit" and set it as a % of total inverter power output, but that affects total inverter power and not just the export to the grid, though the inverter will override that value itself depending on RCRR settings so you may not be able to do what you want.

Altenatively, if you don't want to export at all then you can set the Storage Command Mode to "Discharge to Minimise Import" - the SE definition of this mode is "5 – Discharge to meet loads consumption. Discharging to the grid is not allowed."  However, note that it will revert to the default mode when:
a) the timer runs out
b) the inverter 'resets' on its own for some reason or another

Also, depending on where in the world you are, 15sec may be outside the required reaction time to avoid excess grid output.
